javalanglib/src/main/scala/java/lang/Thread.scala

```diff
@@ -12,12 +12,10 @@
 
 package java.lang
 
-/* We need a constructor to create SingleThread in the companion object, but
- * we don't want user code doing a 'new Thread()' to link, because that could
- * be confusing.
- * So we use a binary signature that no Java source file can ever produce.
+/* We need a constructor to create SingleThread in the companion object, but:
+ * PLEASE DO NOT USE THIS CLASS IN USER CODE!
  */
-class Thread private (dummy: Unit) extends Runnable {
+class Thread extends Runnable {
```

Comment:
I'm afraid this change does not fit in our global policy of "if it links, it is correct". Therefore we cannot accept this change in the core repo.

Reply:
I understand your reason, in defense of this change I would like to say that the current signature is not respecting the contract in the JavaDoc (hence the comment above):
https://docs.oracle.com/javase/7/docs/api/java/lang/Thread.html#Thread()
that allow the construction of the Thread object.

A program that implements `Thread` in a way that is supported by the runtime *is* correct in my opinion, as an example a user can leverage GraalVM and provide support for multi-threading for the specific target; I think that being forced to use alternative classes is a stretch.

This said I "can" keep this file in the Akka.Js codebase (despite the fact that I'm going to provide the users a bad experience in this sense e.g. linking will succeed or fail depending on your dependencies) but I'm not able to access sibling classes therefore this hack:
https://github.com/akka-js/akka.js/blob/sjs1Support/akka-js-actor/js/src/main/scala/java/lang/Thread.scala#L38-L40

Is possibly going to have deeper consequences, can you have a look @sjrd ?

Reply:
Another approach that I would like to explore with you is to make the `Thread` constructor package `protected`.
This way:
 - normal users are not allowed to instantiate it
 - edge cases and/or ppl that are aware of the limitations etc. etc. are able to workaround

Reply:
The problem with all these solutions is always the same: it makes it possible to link a program that *doesn't* use any weird stuff, make it link, and then expect it to work but it won't! This goes against our policy. It doesn't matter that *if* you carefully override all the things to make it work *then* you have a correct program.

The existing constructor of `java.lang.Thread` is `private` and intentionally uses a different signature *so that it won't link*, because it is not correct in general. It is only correct to the main and only thread of a JS environment.

I am afraid that, by trying too hard to reuse `java.lang.Thread` as an interface, you *are* looking for trouble because you try to repurpose JDK APIs with different specifications. This goes against the policy of Scala.js and our implementation of the JDK. Instead, you should abstract these things away as a separate interface, so that there are contractually valid implementations of *that* interface on both platforms.

Reply:
I double checked and I should be able to keep this class in akka.js codebase.
Just need to confirm with you that: `thread.getStackTrace` is not used for functional reasons by Scala.Js itself (at a first look it doesn't looks like).

Reply:
It's only used by `thread.printStackTrace`.

Reply:
Fine, so I decline this PR